avocado/core/remoter.py

```diff
@@ -70,6 +71,10 @@ def __init__(self, hostname, username=None, password=None,
         self.password = password
         self.port = port
         self.quiet = quiet
+        reject_unknown_hosts = settings.get_value('remoter.behavior',
+                                                  'reject_unknown_hosts',
+                                                  key_type=bool,
+                                                  default=False)
```

Comment:
Hmm as we want to support default as `True`, how about using it here and putting the safe default only in config? (we might forget to update this value when the bug is resolved)